[Mahidharah] iP

Mahidharah opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Mahidharah committed:
1. New chatbot name
2. Chatbot simply greets and exits

Mahidharah committed:
1. Echos commands entered by the user, and exits when the user types the command bye

Mahidharah committed:
1. Store whatever text entered by the user
2. Display them back to the user when requested with "list" command

Mahidharah committed:
1. Added the ability to mark tasks as done
2. New class to encapsulate a task and its functions

Mahidharah committed:
Added support for tracking three types of tasks: ToDos, Deadlines and Events

Mahidharah committed:
Testing

Mahidharah committed:
no message

Mahidharah committed:
no message

Mahidharah committed:
Enabled saving the tasks in the hard disk automatically whenever the task list changes and loading the data from the hard disk when the chatbot starts up. File name is hardcoded and relative path from the project root e.g., ./data/duke.txt. If corrupted, error msg displayed

Mahidharah committed:
Enabled storing of deadline and event dates as a java.time.LocalDate in your task objects. Accept dates in a format such as yyyy-mm-dd format (e.g., 2019-10-15) and print in a different format such as MMM dd yyyy e.g., (Oct 15 2019).

Mahidharah committed:
no message

TerrenceFyx commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,158 @@
+            System.out.println("error");
+        }
+    }
+    public static void run(File file, String userinput, ArrayList<Task> list) throws DukeException {
```
Method is  abit long, could consider splitting it

TerrenceFyx commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+public abstract class Task {
+    private boolean done;
```
```
boolean isDone
```

TerrenceFyx commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        return this.done;
+    }
+
+    public String getname() {
```
Good naming of getter

TerrenceFyx reviewed (commented):
LGTM

Mahidharah committed:
Added OOP to code

Mahidharah committed:
segmented code into packages

mingyu-wan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,158 @@
+        }
+    }
+
+    public static void cansplit(String input, String splitter) throws DukeException {
```
could camelCase the method

mingyu-wan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,158 @@
+        }
+    }
+
+    public static void runfile(File file, String datainput, ArrayList<Task> list) throws DukeException{
```
could camelCase the method

mingyu-wan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,158 @@
+            return;
+
+        } else if (userinput.length() >= 6 && userinput.substring(0, 5).equalsIgnoreCase("mark ")) {
+            try {
```
Could standardize if there is a new line for instructions after the if else

mingyu-wan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,158 @@
+import java.io.*;
```
Could be specific on what you want to import

mingyu-wan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public String toString() {
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM d yyyy");
+        return "[E]" + super.toString() + " (from: " + this.from.format(format) +
```
Good division of content and not leaving it as one long line

mingyu-wan reviewed (commented):
Generally lgtm, coding standards were followed just some small potential problems. Could be better if there was javadocs to each of the methods for better understanding.

Mahidharah committed:
Added packages

Mahidharah committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

* commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552':
  Bump gradle and lib version
  Add Gradle support

Mahidharah committed:
failed attempt at JUnit and gradle

Mahidharah committed:
Gradle fixed and Junit tests for 1 class completed

Mahidharah committed:
Jar?

Mahidharah committed:
Javadocs added

Mahidharah committed:
Added ability to find tasks with keyword

Mahidharah committed:
checkedstyle

Mahidharah committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  checkedstyle

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/parsers/Parser.java

Mahidharah committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Added ability to find tasks with keyword

# Conflicts:
#	src/main/java/duke/parsers/Parser.java
#	src/main/java/duke/tasks/Task.java

Mahidharah committed:
fixed bugs

Mahidharah committed:
gui implementation that does not work

Mahidharah committed:
Updated README.md

Mahidharah committed:
Updated README.md

Mahidharah committed:
Update README.md

Mahidharah committed:
Working GUI prototype, as per step 3 of tutorial. Bugs: 1. Just pressing enter returns a msg. 2. Lines not indented, as of not %n placeholder suggests lines. 3. Cannot exit program

Mahidharah committed:
Merge commit '2cc448446b39a78354ee178a8177ed2e99bbb755'

* commit '2cc448446b39a78354ee178a8177ed2e99bbb755':
  Update README.md
  Updated README.md
  Updated README.md

Mahidharah committed:
Style was all over the place

Naming and javadocs did not follow perfectly follow style guidelines

Incorporated checkstyle to aid with fixing style in code

Mahidharah committed:
Parse method was too long.

Such methods violate code quality and should be avoided.

Broke up methods separately to show clearly what is happening at each step and used switch and case instead.

This allows easy reading and understanding of the method.

Old methods commented in for reference.

Mahidharah committed:
Fix codestyle

Code was not neat and conventions were not followed.

Made changes to names and whitespaces using check-style, but some errors are unavoidable.

Mahidharah committed:
Merge pull request #2 from Mahidharah/branch-A-CodeQuality

Branch a code quality

Mahidharah committed:
Added assertions based on assumptions.

Assertions help detect errors due to assumptions.

Added assertions by examining all methods to see where explicit assumptions have been made.

Mahidharah committed:
Added schedule command.

Allows user to check time sensitive tasks if the tasks occur on the given date with the schedule command.

Only applicable for Events and Deadlines, and only shows deadlines on the day and events where the date is in the range, inclusive of from and to.

Mahidharah committed:
Improved GUI.

A better GUI which is more aesthetically pleasing helps make project look more professional

Added necessary padding for better visuals. Made images circular for enhanced appearance. Used javafx Circle instead of ImageView.

Image provided will need to be square, otherwise image will be resized to fit the circle.

Mahidharah committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Improved GUI.

Mahidharah committed:
Improve GUI

Help make look GUI cleaner and more professional with boxes for labels.

Created grey boxes for labels

Mahidharah committed:
no message

Mahidharah committed:
Create Ui.png

Mahidharah committed:
Add files via upload

Mahidharah committed:
Delete docs/iP_screenshot.png

Mahidharah committed:
Delete docs/Ui.png

Mahidharah committed:
Add files via upload

Mahidharah committed:
Update README.md

Mahidharah committed:
Update README.md

Mahidharah committed:
Generated JAR file

Mahidharah committed:
Merge commit '05e59794da7976c6e995881a5502bfcee76c343f'

* commit '05e59794da7976c6e995881a5502bfcee76c343f':
  Update README.md
  Update README.md
  Add files via upload
  Delete docs/Ui.png
  Delete docs/iP_screenshot.png
  Add files via upload
  Create Ui.png
